lib/rubocop/cop/style/symbol_proc.rb

```diff
@@ -22,11 +22,12 @@ class SymbolProc < Base
         SUPER_TYPES = %i[super zsuper].freeze
 
         def_node_matcher :proc_node?, '(send (const {nil? cbase} :Proc) :new)'
+        def_node_matcher :symbol_proc_receiver?, '{(send ...) (super ...) zsuper}'
         def_node_matcher :symbol_proc?, <<~PATTERN
-          ({block numblock}
-            ${(send ...) (super ...) zsuper}
-            ${(args (arg _)) 1}
-            (send (lvar _var) $_))
+          {
+            (block $#symbol_proc_receiver? $(args (arg _var)) (send (lvar _var) $_))
+            (numblock $#symbol_proc_receiver? $1 (send (lvar :_1) $_))
```

Comment:
I made a double take at `$1`! 😆

Reply:
Yeah it’s needed to balance each option in the `{}` but not actually used for anything (the capture for `block` is though so it’s needed)

Reply:
That's what I figured 😆